### DISCUSSION TITLE
[Discuss] Dropping `default_short_url` column in favor of `Accessor`

### DISCUSSION THREAD

**Discussion by iRaziul** [ash-jc-allen/short-url#248] [Ideas]
This package stores the full URL in the database (`default_short_url`) but I have a few thoughts.
My idea is to drop that column and utilize `Accessor` (Dynamic attribute).

### Why?
- Storing the full URL in the database will increase database size.
- When the config value of `short-url.prefix` is changed, `default_short_url` for the existing links is not updated. (Same for the domain)

So, utilizing the accessor we can return `default_short_url` from the model and the mentioned cases will be solved as well.

![database screenshot](https://github.com/ash-jc-allen/short-url/assets/51883557/4b4205d6-f83a-40e4-bfd7-aeaaa6df42eb)

**Comment by ash-jc-allen** [comment]
Hey @iRaziul, thanks for the suggestion! I think if I built the package again from the ground up, I probably wouldn't add the `default_short_url` column in the database. So I do agree with you on that! 😄

But I think I might hold off on adding something like this because I'm still happy with the existing functionality. It's all personal preference, but I kinda like the fact that the `default_short_url` holds a hardcoded URL that was set at the time of creation. I also imagine that it's not very often (or likely) that the short URL key or domain will change much after creating the short URLs.

If enough people think this is something worth looking at though, I'll definitely consider it! 🙂
